app/controllers/api/notifications_controller.rb

```diff
@@ -0,0 +1,48 @@
+class Api::NotificationsController < ActionController::Base
+  before_action :set_raven_context
+  rescue_from ActionController::ParameterMissing, with: :render_unprocessable_entity_response
+
+  def create
+    bot = Bot.find_by_notifications_secret(params[:notifications_secret])
+    return render json: {error: "Unknown notification secret"}, status: :unauthorized unless bot
+
+    @notification = bot.notifications.create! notification_params
+
+    if @notification.save
+      render json: {result: :ok}
+    else
+      render json: @notification.errors, status: :unprocessable_entity
+    end
+  end
+
+  private
+    def notification_params
+      the_params = params.require(:notification).permit(:type)
+      the_params[:data] = params.require(:notification).fetch(:data, {}).permit!
+      the_params
+    end
+
+    def render_unprocessable_entity_response(exception)
+      render json: 'Invalid notification', status: :unprocessable_entity
+    end
+
+    def set_raven_context
```

Comment:
Instead of setting raven context here, we could inherit from ApplicationController turning off request forgery protection. By adding to our class one of these: protect_from_forgery except: :create | skip_forgery_protection.

From: http://api.rubyonrails.org/classes/ActionController/RequestForgeryProtection/ClassMethods.html